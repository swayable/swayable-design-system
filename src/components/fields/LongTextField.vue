<template>
  <div class='flex flex-col'>
    <p class='text-primary typography-5 px-0.5'>
      {{ prompt }}
    </p>
    <p
      v-if='helpText'
      class='text-tertiary typography-8 px-0.5'
    >
      {{ helpText }}
    </p>
    <TextArea
      :id='questionId'
      :placeholder='placeholder'
      :name='name'
      :required='isRequired'
      class='flex-grow mt-1'
      v-on='$listeners'
    />
  </div>
</template>
<script>
import BaseField from './BaseField'
export default {
  name:'LongTextField',
  extends: BaseField,
  props: {
    placeholder: {
      type: String,
      default: 'Answer',
    },
  },
}
</script>

<docs>
```jsx
  let inputText
  <div class='p-2'>
    <LongTextField
      id='1'
      name='example'
      :required='true'
      prompt="How much do you love this input?"
      v-model='inputText'
    />
    <span> {{ inputText }} </span>
  </div>
```
<docs>