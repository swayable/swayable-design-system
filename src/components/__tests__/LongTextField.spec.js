import { shallowMount, mount } from '@vue/test-utils'
import LongTextField from '../fields/LongTextField.vue'

const exampleProps = {
  id: 1,
  name: 'example',
  required: true,
  placeholder: 'enter text here',
  prompt: 'How much do you love this input?',
}

describe('match the snapshot', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(LongTextField, { propsData: exampleProps })
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.destroy()
  })
})


describe('help text passed in', () => {
  it('renders helpText only when provided', () => {
    const helpText = 'Help!'
    let wrapper = mount(LongTextField, { propsData: exampleProps })
    
    expect(wrapper.text()).not.toContain(helpText)

    wrapper = mount(LongTextField, { propsData: { helpText, ...exampleProps }})
    expect(wrapper.text()).toContain(helpText)
  })
})

