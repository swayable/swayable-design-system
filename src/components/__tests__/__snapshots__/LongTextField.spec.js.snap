// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot renders correctly 1`] = `
<div class="flex flex-col" id="1">
  <p class="text-primary typography-5 px-0.5">
    How much do you love this input?
  </p>
  <!---->
  <textarea-stub placeholder="enter text here" name="example" class="flex-grow mt-1"></textarea-stub>
</div>
`;
